Pcap++/header/PcapLiveDeviceList.h

```diff
@@ -54,48 +54,48 @@ namespace pcpp
 		/**
 		 * @return A vector containing pointers to all live devices currently installed on the machine
 		 */
-		inline const std::vector<PcapLiveDevice*>& getPcapLiveDevicesList() { return m_LiveDeviceList; }
+		const std::vector<PcapLiveDevice*>& getPcapLiveDevicesList() const { return m_LiveDeviceList; }
 
 		/**
 		 * Get a pointer to the live device by its IP address. IP address can be both IPv4 or IPv6
 		 * @param[in] ipAddr The IP address defined for the device
 		 * @return A pointer to the live device if this IP address exists. NULL otherwise
 		 */
-		PcapLiveDevice* getPcapLiveDeviceByIp(IPAddress* ipAddr);
+		PcapLiveDevice* getPcapLiveDeviceByIp(IPAddress* ipAddr) const;
 
 		/**
 		 * Get a pointer to the live device by its IPv4 address
 		 * @param[in] ipAddr The IPv4 address defined for the device
 		 * @return A pointer to the live device if this IPv4 address exists. NULL otherwise
 		 */
-		PcapLiveDevice* getPcapLiveDeviceByIp(IPv4Address ipAddr);
+		PcapLiveDevice* getPcapLiveDeviceByIp(IPv4Address ipAddr) const;
 
 		/**
 		 * Get a pointer to the live device by its IPv6 address
 		 * @param[in] ip6Addr The IPv6 address defined for the device
 		 * @return A pointer to the live device if this IPv6 address exists. NULL otherwise
 		 */
-		PcapLiveDevice* getPcapLiveDeviceByIp(IPv6Address ip6Addr);
+		PcapLiveDevice* getPcapLiveDeviceByIp(IPv6Address ip6Addr) const;
 
 		/**
 		 * Get a pointer to the live device by its IP address represented as string. IP address can be both IPv4 or IPv6
 		 * @param[in] ipAddrAsString The IP address defined for the device as string
 		 * @return A pointer to the live device if this IP address is valid and exists. NULL otherwise
 		 */
-		PcapLiveDevice* getPcapLiveDeviceByIp(const char* ipAddrAsString);
+		PcapLiveDevice* getPcapLiveDeviceByIp(const char* ipAddrAsString) const;
 
 		/**
 		 * Get a pointer to the live device by its name
 		 * @param[in] name The name of the interface (e.g eth0)
 		 * @return A pointer to the live device if this name exists. NULL otherwise
 		 */
-		PcapLiveDevice* getPcapLiveDeviceByName(const std::string& name);
+		PcapLiveDevice* getPcapLiveDeviceByName(const std::string& name) const;
 
 		/**
 		 * @return A list of all DNS servers defined for this machine. If this list is empty it means no DNS servers were defined or they
 		 * couldn't be extracted from some reason
 		 */
-		std::vector<IPv4Address>& getDnsServers();
+		std::vector<IPv4Address>& getDnsServers() { return m_DnsServers; } // TODO: Should this method really return a non-const reference?

```

Comment:
Pay attention on comment

Reply:
This list is read-only and shouldn't be changed by the caller so I guess the answer to your question is "no" - we should probably make it a const reference